src/as2fm/scxml_converter/xml_data_types/xml_types.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2025 - for information on the respective copyright owner
+# see the NOTICE file
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List
+
+import lxml.etree as ET
+
+from as2fm.as2fm_common.common import remove_namespace
+from as2fm.as2fm_common.logging import get_error_msg, set_filepath_for_all_elements
+from as2fm.scxml_converter.xml_data_types.xml_struct_definition import XmlStructDefinition
+
+
+def read_types_file(
+    xml_path: str,
+) -> List[XmlStructDefinition]:
```

Comment:
This could return a dictionary, since we are using that throughout the whole pipeline